go/vt/vttablet/tabletserver/schema/db.go

```diff
@@ -86,6 +88,32 @@ where table_schema = database() and table_name in ::viewNames`
 
 	// fetchTablesAndViews queries fetches all information about tables and views
 	fetchTablesAndViews = `select table_name, create_statement from %s.tables where table_schema = database() union select table_name, create_statement from %s.views where table_schema = database()`
+
+	// detectUdfChange query detects if there is any udf change from previous copy.
+	detectUdfChange = `SELECT name
+FROM (
+	SELECT name FROM 
+	mysql.func 
+
+	UNION ALL
+
+	SELECT function_name
+	FROM %s.udfs
+) _inner
+GROUP BY name
+HAVING COUNT(*) = 1
+LIMIT 1
+`
+
+	// deleteAllUdfs clears out the udfs table.
+	deleteAllUdfs = `delete from %s.udfs`
+
+	// copyUdfs copies user defined function to the udfs table.
+	copyUdfs = `INSERT INTO %s.udfs(FUNCTION_NAME, FUNCTION_RETURN_TYPE, FUNCTION_TYPE) 
+SELECT f.name, i.UDF_RETURN_TYPE, f.type FROM mysql.func f left join performance_schema.user_defined_functions i on f.name = i.udf_name
```

Comment:
nit: I assume you use `i` as an alias for `information_schema`, so it makes sense to use `p` for `performance_schema`